wire-ios/WireUITests/Pages/PageModel.swift

```diff
@@ -19,14 +19,32 @@
 import XCTest
 
 class PageModel {
+
+    enum Failure: Error {
+        case notLoaded(PageModel)
+        
+        var localizedDescription: String {
+            switch self {
+            case .notLoaded(let pageModel):
+                return "Page \(String(describing: pageModel)) not loaded"
+            }
+        }
+    }
+
     let app: XCUIApplication
 
     init() {
         self.app = XCUIApplication()
-        hasLoaded()
     }
 
-    func hasLoaded() {
-        print("Warning: hasLoaded should be implemented by inheritor")
+    var pageMainElement: XCUIElement {
+        fatalError("override this in subclass \(String(describing: self))")
```

Comment:
here's it's ok to use fatalError as it's a programmatic error not to override this in subclass